docs/data/material/experimental-api/css-variables/css-variables.md

```diff
@@ -0,0 +1,128 @@
+# CSS variables
+
+<p class="description">Learn about the experimental API for using CSS variables on Material UI components.</p>
+
+CSS variables is the most upvoted `@mui/system` related issue.
+It allows several improvements in developer experience, like fixing dark mode SSR flickering, allowing multiple themes other than only light and dark, better debugging experience overall, and many others.
+To learn more, check [the RFC](https://github.com/mui/material-ui/issues/27651) we did about it.
+
+A while ago, we added an experimental API to the System package meant to allow styling the components with CSS variables.
+We want now to take it one step further and utilize it in one of the Material UI components.
+
+## Usage
+
+At this moment, the API does not require any breaking changes for the Material UI components, however it depends on using a new experimental provider for the theme, called `Experimental_CssVarsProvider`.
+Except for providing the theme in the inner React context, this new provider has as a responsibility to generate CSS variables out of all tokens in the theme that are not functions, and make them available in the context.
+
+All these variables, are available under a key in the theme, named `vars`.
+The structure of this object is identical to the theme structure, the only difference is that the values represent some CSS varaibles.
+
+The best way to see this is by example.
+Let's say we have this element that uses some theme token:
+
+```jsx
+const Example() {
+  const theme = useTheme();
+  return <div style={{ color: theme.palette.success.main }}>Success div</div>
+}
+```
+
+{{"demo": "SuccessDivThemeToken.js", "hideToolbar": true }}
+
+If you try to inspect it, you will see that the color style has a direct hex value, that comes from some theme token:
+
+```html
+<div style="color: #2e7d32;">...</div>
+```
+
+Now, if the same div used CSS variables, as in the example below, you'll be able to track where that color token is defined in the theme:
+
+```jsx
+const Example() {
+  const theme = useTheme();
+  return <div style={{ color: theme.vars.palette.success.main }}>Success div</div>
+}
+```
+
+{{"demo": "SuccessDivCSSVariable.js", "hideToolbar": true }}
+
+```html
+<div style="color: var(--md-palette-success-main);">...</div>
+```
+
+Additionally, by clicking on the CSS variable in the browser's dev tools, you'll be taken to the theme object where you'll see its direct hex value again.
+We believe this to be a significant improvement in developer experience when debugging styles for any component.
+
+### Creating a custom theme
+
+The example above is using the default Material UI theme.
+But, what if you wanted to create your own custom theme?
+Let's see how to do that using the `experimental_extendTheme` utility.
```

Comment:
Let's not assume that the reader has read the previous section. They might just be skimming.

```suggestion
If you want to override Material UI's default theme, you can create your own custom theme with CSS variables using the `experimental_extendTheme` utility.
```